### PR TITLE
Update --sink flag description

### PR DESCRIPTION
**Pull request by navidshaikh**
## Description
 for referring to a Knative service in another namespace by
 'ksvc' prefix explicitly, for e.g. 'ksvc:name:namespace'

## Changes
Update --sink flag description

## Reference
 xref: https://github.com/knative/client/pull/1264

/cc @rhuss @dsimansk

### CODE REVIEW

File: pkg/kn/commands/flags/sink.go [hunk 1]
```diff
@@ -107,7 +108,7 @@ func (i *SinkFlags) ResolveSink(knclient clientdynamic.KnDynamicClient, namespac
 		if prefix == "svc" || prefix == "service" {
 			return nil, fmt.Errorf("unsupported sink prefix: '%s', please use prefix 'ksvc' for knative service", prefix)
 		}
-		return nil, fmt.Errorf("unsupported sink prefix: '%s'", prefix)
+		return nil, fmt.Errorf("unsupported sink prefix: '%s', If referring to a knative service in another namespace, 'ksvc:name:namespace' combination must be provided explicitly", prefix)
```
Comment (dsimansk) [inline pkg/kn/commands/flags/sink.go:111]:
```suggestion
		return nil, fmt.Errorf("unsupported sink prefix: '%s', if referring to a Knative service in another namespace, 'ksvc:name:namespace' combination must be provided explicitly", prefix)
```

File: docs/cmd/kn_source_container_update.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ kn source container update NAME --image IMAGE
...
+  -s, --sink string              Addressable sink for events. You can specify a broker, channel, Knative service or URI. Examples: '--sink broker:nest' for a broker 'nest', '--sink channel:pipe' for a channel 'pipe', '--sink ksvc:mysvc:mynamespace' for a Knative service 'mysvc' in another namespace 'mynamespace', '--sink https://event.receiver.uri' for an URI with an 'http://' or 'https://' schema, '--sink ksvc:receiver' or simply '--sink receiver' for a Knative service 'receiver' in the current namespace. If a prefix is not provided, it is considered as a Knative service in the current namespace. If referring to a Knative service in another namespace, 'ksvc:name:namespace' combination must be provided explicitly.
```
Comment (maximilien) [inline docs/cmd/kn_source_container_update.md:33]: Cool but long :(

File: pkg/kn/commands/flags/sink.go [hunk 3]
```diff
@@ -107,7 +108,7 @@ func (i *SinkFlags) ResolveSink(knclient clientdynamic.KnDynamicClient, namespac
 		if prefix == "svc" || prefix == "service" {
 			return nil, fmt.Errorf("unsupported sink prefix: '%s', please use prefix 'ksvc' for knative service", prefix)
 		}
-		return nil, fmt.Errorf("unsupported sink prefix: '%s'", prefix)
+		return nil, fmt.Errorf("unsupported sink prefix: '%s', if referring to a knative service in another namespace, 'ksvc:name:namespace' combination must be provided explicitly", prefix)
```
Comment (maximilien) [inline pkg/kn/commands/flags/sink.go:111]: Shouldn’t `if` really be `when`?

**Review by maximilien — COMMENTED** [review]
Left couple minor comments...

File: docs/cmd/kn_source_container_update.md (hunk 2 above)
Comment (navidshaikh) [reply docs/cmd/kn_source_container_update.md:33]: yes, but sink flag takes multiple combinations and it's better to be explicit with examples IMO
I'll also look to contribute some examples for this in the docs

File: pkg/kn/commands/flags/sink.go (hunk 3 above)
Comment (navidshaikh) [reply pkg/kn/commands/flags/sink.go:111]: for the special case of knative service, where we may not require a prefix, we cant detect if its ksvc name or a prefix (defined in config)

**Comment by maximilien** [comment]
Ok cool.
/approve
/hold for others to chime in

**Comment by dsimansk** [comment]
@navidshaikh @maximilien fyi removing hold to get the PR merged given there's no more comments.


/unhold

**Pull request merged**